src/pseudopeople/interface.py

```diff
@@ -64,13 +67,17 @@ def _generate_dataset(
         if len(data_paths) > 1
         else data_paths
     )
+
     for data_path in iterator:
         logger.debug(f"Loading data from {data_path}.")
-        data = _load_data_from_path(data_path, user_filters)
-        if data.empty:
-            continue
-        data = _reformat_dates_for_noising(data, dataset)
-        data = _coerce_dtypes(data, dataset)
+        if dataset.name == DatasetNames.TAXES_1040:
```

Comment:
I think it's cleaner to move the logic to differentiate between the 1040 and the other datasets inside the method `_load_data_from_path`.

Reply:
I'm not 100% sure I understand what I should do here for this. If it's the 1040 it calls a different function vs all other datasets call `_load_data_from_path`. Should I do this check in there then call the separate methods in there? This would conceptually mean `_load_data_from_path `loads and preps the data for `noise_dataset `which I think is pretty nice but would be a bit confusing for additional methods to be called in there and not this main runner loop.